pkg/objstore/client/factory.go

```diff
@@ -76,5 +76,11 @@ func NewBucket(logger log.Logger, confContentYaml []byte, reg prometheus.Registe
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("create %s client", bucketConf.Type))
 	}
-	return objstore.BucketWithMetrics(bucket.Name(), bucket, reg), nil
+
+	var bucketReg prometheus.Registerer
+	if reg != nil {
```

Comment:
Actually you can just pass it, no need for check IF you will use `exthttp.WrapRegisterer..`

Reply:
Ups, right. Thanks, done!